python/pyspark/context.py

```diff
@@ -327,24 +360,24 @@ def _do_init(
             dump_path = self._conf.get("spark.python.profile.dump", None)
             self.profiler_collector = ProfilerCollector(profiler_cls, udf_profiler_cls, dump_path)
         else:
-            self.profiler_collector = None
+            self.profiler_collector = None  # type: ignore[assignment]
 
         # create a signal handler which would be invoked on receiving SIGINT
-        def signal_handler(signal, frame):
+        def signal_handler(signal: Any, frame: Any) -> NoReturn:
```

Comment:
Should we look at this? 

https://github.com/apache/spark/pull/34466#discussion_r753823799

Reply:
If we copy the class declaration from typeshed, it seems that it will add a lot of codes. Is there a better way to achieve this?

Reply:
I was thinking more about narrowing down the types than direct copy and paste, but it is probably OK to keep it as-is for now ‒ it is deep internals anyway, and unlikely to be silently broken.